blocks/gmo-campaign-list/gmo-campaign-list.js

```diff
@@ -123,14 +123,33 @@ const testConfig = [
     }
 ]
 
+//Global variables used by helper functions
+let currentPageInfo = {};
+let cursorArray = [];
+let currentPage = 1;
+let currentNumberPerPage = 4
+//Get Campaign Count for pagination
+const campaignCount = await graphqlCampaignCount();
+
+export default async function decorate(block, numPerPage = currentNumberPerPage, cursor = '', previousPage = false, nextPage = false) {
+
+    const campaignPaginatedResponse = await graphqlAllCampaigns(numPerPage, cursor);
+    const campaigns = campaignPaginatedResponse.data.campaignPaginated.edges;
+    currentPageInfo = campaignPaginatedResponse.data.campaignPaginated.pageInfo;
+
+    currentPageInfo.nextCursor = campaigns[campaigns.length - 1].cursor;
+    if (!previousPage && !nextPage)
+    {
+      cursorArray = campaigns.map(item => item.cursor);
+    }
+    else if (nextPage){
+
+      campaigns.forEach(item => {
+          cursorArray.push(item.cursor);
+      });
+    }
+    currentPageInfo.itemCount = campaigns.length;
 
-export default async function decorate(block) {
-    const campaignResponse = await graphqlAllCampaigns();
-    console.log(campaignResponse);
-    const campaigns = campaignResponse.data.campaignList.items;
-    console.log(campaignResponse.data.campaignList.items);
-    const numPerPage = 4; 
-    const campaignCount = campaigns.length;
     const config = testConfig;
```

Comment:
remove `testConfig`

Reply:
I think testConfig is used to provide test/dummy data at the moment for the listHeaders, and will be replaced by data from graphql in future Workfront Task